Dockerfile

```diff
@@ -44,94 +54,52 @@ RUN set -eux; \
 	chown redmine:redmine "$HOME"; \
 	chmod 1777 "$HOME"
 
-ARG REDMINE_VERSION="4.2.3"
-ARG REDMICA_VERSION=""
-# ENV REDMINE_DOWNLOAD_SHA256 ad4109c3425f1cfe4c8961f6ae6494c76e20d81ed946caa1e297d9eda13b41b4
+# To support Redmine 5.0
+ENV GEM_RGEO_ACTIVERECORD_VERSION=7.0.1
+ENV GEM_ACTIVERECORD_POSTGIS_ADAPTER_VERSION=7.1.1
+
+# Defined in docker-compose.yml
+ARG REDMINE_VERSION
+ARG REDMINE_DOWNLOAD_SHA256
 
 RUN set -eux; \
-	if [ -n "$REDMINE_VERSION" ]; then \
-		wget -O redmine.tar.gz "https://www.redmine.org/releases/redmine-${REDMINE_VERSION}.tar.gz"; \
-		# echo "$REDMINE_DOWNLOAD_SHA256 *redmine.tar.gz" | sha256sum -c -;
-	elif [ -n "$REDMICA_VERSION" ]; then \
-		wget -O redmine.tar.gz "https://github.com/redmica/redmica/archive/v${REDMICA_VERSION}.tar.gz"; \
-	fi; \
+  # bullseye hack
+	wget --no-check-certificate -O redmine.tar.gz "https://www.redmine.org/releases/redmine-${REDMINE_VERSION}.tar.gz"; \
+	echo "${REDMINE_DOWNLOAD_SHA256} *redmine.tar.gz" | sha256sum -c -; \
 	tar -xf redmine.tar.gz --strip-components=1; \
 	rm redmine.tar.gz files/delete.me log/delete.me; \
-	mkdir -p log public/plugin_assets sqlite tmp/pdf tmp/pids; \
-	chown -R redmine:redmine ./; \
-# log to STDOUT (https://github.com/docker-library/redmine/issues/108)
-	echo 'config.logger = Logger.new(STDOUT)' > config/additional_environment.rb; \
-# fix permissions for running as an arbitrary user
-	chmod -R ugo=rwX config db sqlite; \
-	find log tmp -type d -exec chmod 1777 '{}' +
-
-# for Redmine patches
-ARG PATCH_STRIP=1
-ARG PATCH_DIRS=""
-COPY patches/ ./patches/
-
-# for GTT gem native extensions
-ARG GEM_PG_VERSION="1.2.3"
-COPY Gemfile.local ./
-COPY plugins/ ./plugins/
+	mkdir -p log public/plugin_assets tmp/pdf tmp/pids; \
+	chown -R redmine:redmine ./ ; \
+  chmod -R ugo=rwX config ; \
+  find log tmp -type d -exec chmod 1777 '{}' +
+
+COPY --from=gtt-builder --chown=redmine:redmine /app/ ./plugins/
+
+COPY --chown=redmine:redmine config/ ./config/
+
+COPY --chown=redmine:redmine Gemfile.local ./
 
 RUN set -eux; \
 	\
 	savedAptMark="$(apt-mark showmanual)"; \
 	apt-get update; \
 	apt-get install -y --no-install-recommends \
-		# freetds-dev \
+		freetds-dev \
```

Comment:
`freetds-dev` is for MS SQL Server client library, so just commenting out or deleting line may be better.
https://packages.debian.org/en/sid/freetds-dev